kroxylicious-systemtests/src/test/java/io/kroxylicious/systemtests/filters/AuthorizationST.java

```diff
@@ -0,0 +1,229 @@
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assumptions.assumeThat;
+import static org.junit.jupiter.api.Assertions.assertAll;
+
+class AuthorizationST extends AbstractST {
+    protected static final String BROKER_NODE_NAME = "kafka";
+    private static final Logger LOGGER = LoggerFactory.getLogger(AuthorizationST.class);
+    private static final String MESSAGE = "Hello-world";
+    private final String clusterName = "my-cluster";
+    private String bootstrap;
+    private KroxyliciousOperator kroxyliciousOperator;
+    private static Kroxylicious kroxylicious;
+    private final Map<String, String> usernamePasswords = new HashMap<>();
+    List<String> aclRules;
+
+    @BeforeAll
+    void setUp() {
+        List<Pod> kafkaPods = kubeClient().listPodsByPrefixInName(Constants.KAFKA_DEFAULT_NAMESPACE, clusterName);
+        if (!kafkaPods.isEmpty()) {
+            LOGGER.atInfo().setMessage("Skipping kafka deployment. It is already deployed!").log();
+        }
+        else {
+            LOGGER.atInfo().setMessage("Deploying Kafka in {} namespace").addArgument(Constants.KAFKA_DEFAULT_NAMESPACE).log();
+
+            int numberOfBrokers = 1;
+            KafkaBuilder kafka = KafkaTemplates.kafkaPersistentWithAuthentication(Constants.KAFKA_DEFAULT_NAMESPACE, clusterName, numberOfBrokers);
+
+            resourceManager.createResourceFromBuilderWithWait(
+                    KafkaNodePoolTemplates.kafkaBasedNodePoolWithDualRole(BROKER_NODE_NAME, kafka.build(), numberOfBrokers),
+                    kafka);
+        }
+
+        kroxyliciousOperator = new KroxyliciousOperator(Constants.KROXYLICIOUS_OPERATOR_NAMESPACE);
+        kroxyliciousOperator.deploy();
+
+        generatePasswordForNewUser(Constants.KROXYLICIOUS_ADMIN_USER);
+    }
+
+    @BeforeEach
+    void beforeEach() {
+        bootstrap = null;
+        aclRules = new ArrayList<>();
+        aclRules.add(generateAllowAclRule(Constants.KROXYLICIOUS_ADMIN_USER, "*"));
+    }
+
+    @AfterAll
+    void cleanUp() {
+        kroxyliciousOperator.delete();
+    }
+
+    @Test
+    void testScramSha512AuthenticationAllowProduceAndConsume(String namespace) {
+        // kcat does not support scram-sha-512 authentication: https://github.com/edenhill/kcat/issues/462
+        assumeThat(Environment.KAFKA_CLIENT).isNotEqualToIgnoringCase(KafkaClientType.KCAT.name());
+
+        int numberOfMessages = 1;
+        String user = "bob"; // name shall be always lowercase
+        generatePasswordForNewUser(user);
+        aclRules.add(generateAllowAclRule(user, topicName));
+
+        // start Kroxylicious
+        LOGGER.atInfo().setMessage("Given Kroxylicious in {} namespace with {} replicas").addArgument(namespace).addArgument(1).log();
+        kroxylicious = new Kroxylicious(namespace);
+        kroxylicious.deployPortIdentifiesNodeWithAuthorizationFilter(clusterName, usernamePasswords, aclRules);
```

Comment:
We shouldn't need to deploy a new kroxylicious instance for each test case. We can update the ACL rules in each test and the operator will restart the pod automatically.

We should be able to wait for that roll out to happen before progressing the tests.

However I think that deserves its own PR for the following reasons.
-  probably applies to quite a lot of tests 
- Is just an optimisation (at least I'd hope it was faster)
- quite an involved fix.

Reply:
New issue created for that. Thanks! #3038